client/src/pages/BidderSignUp.js

```diff
@@ -0,0 +1,131 @@
+
+	const validateEmail = (email) => {
+		//Regex to check email format is valid
+		const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+		return emailRegex.test(email);
+	};
+
+	async function postBidderDeatils(userData) {
+		try {
+			const response = await fetch("/mock-api", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(userData),
+			});
+			if (!response.ok) {
+				throw new Error("Network response was not ok");
+			}
+			setRegisterStatus("Registered successfully!");
+		} catch (error) {
+			setRegisterStatus("Internal server error!");
+		}
+	}
+
+	const handleInputChange = (e) => {
+		const { name, value } = e.target;
+		setBidderDetails((prevDetails) => ({ ...prevDetails, [name]: value }));
+	};
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		const isEmailValid = validateEmail(bidderDetails.email);
+		if (isEmailValid) {
+			setEmailError("");
+			postBidderDeatils(bidderDetails);
+		} else {
+			if (!validateEmail(bidderDetails.email)) {
+				setEmailError("Enter a valid email address");
+			}
+		}
+	};
+
+	function displayMessage(msg) {
+		if (msg === "successful") {
+			return "registered successfully!";
+		} else if (msg === "invalidEmailFormat") {
+			return "email format is not correct!";
+		} else if (msg === "notRegistered") {
+			return "you are not registered!";
+		} else if (msg === "worngPassword") {
+			return "Password is incorrect!";
+		} else if (msg === "serverError") {
+			return "Internal server error";
+		} else {
+			return "Unkown error";
+		}
+	}
+
+	return (
+		<>
+			<form onSubmit={handleSubmit}>
+				<div>
+					<label htmlFor="first-name">First Name:</label>
+					<input
+						type="text"
+						id="first-name"
+						name="firstName"
+						placeholder="Enter your first name"
+						value={bidderDetails.firstName}
+						onChange={handleInputChange}
+						required
+					/>
+				</div>
+				<div>
+					<label htmlFor="last-name">Last Name:</label>
+					<input
+						type="text"
+						id="last-name"
+						name="lastName"
+						placeholder="Enter your last name"
+						value={bidderDetails.lastName}
+						onChange={handleInputChange}
+						required
+					/>
+				</div>
+				<div>
+					<label htmlFor="user-name">Username:</label>
+					<input
+						type="text"
+						id="user-name"
+						name="userName"
+						placeholder="Enter your username"
+						value={bidderDetails.userName}
+						onChange={handleInputChange}
+						required
+					/>
+				</div>
+				<div>
+					<label htmlFor="email">Email:</label>
+					<input
+						type="email"
+						id="email"
+						name="email"
+						placeholder="Enter your email address"
+						value={bidderDetails.email}
+						onChange={handleInputChange}
+						required
+					/>
+					{emailError && <p>{emailError}</p>}
+				</div>
+				<button type="submit">Submit</button>
+			</form>
+			<div>{displayMessage(registerStatus)}</div>
```

Comment:
You don't need to call a function here. Just make the message correct before rendering the form